src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -239,8 +247,16 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
             PlatformMgr().PostEvent(&advChange);
         }
 
-        // Start timer to disable CHIPoBLE advertisement after timeout expiration
-        SystemLayer.StartTimer(CHIP_DEVICE_CONFIG_BLE_ADVERTISING_TIMEOUT, HandleBLEAdvertisementTimeout, this);
+        // Start timers only in the fast advertising case, in the slow advertising case, they are already started.
+        if (mIsFastAdvertisingOn)
+        {
+            // Start timer to disable CHIPoBLE advertisement after timeout expiration.
+            SystemLayer.StartTimer(CHIP_DEVICE_CONFIG_BLE_ADVERTISING_TIMEOUT, HandleBLEAdvertisementTimeout, this);
```

Comment:
Does it automatically cancel previous timers? Worth checking so they don't stack up.

Reply:
Yes it cancels the currently running timer if you call this method with the same callback and state. If you run another timer for different event it does not cancel it.